src/sync/sync_config.hpp

```diff
@@ -100,6 +104,7 @@ struct SyncConfig {
     SyncSessionStopPolicy stop_policy;
     std::function<SyncBindSessionHandler> bind_session_handler;
     std::function<SyncSessionErrorHandler> error_handler;
+    std::shared_ptr<ChangesetTransformer> transformer;
```

Comment:
Ok, so you are destroying the cooker when the `SharedRealm` is destroyed? This is a problem if it leads to a violation of the rule stated here https://github.com/realm/realm-sync/blob/b10fe25d4477603e3deddf06b8363d250ccd332c/src/realm/sync/client.hpp#L231. I do realize that this can be a pain.